lib/AST/ClangTypeConverter.cpp

```diff
@@ -158,6 +158,57 @@ const clang::Type *ClangTypeConverter::getFunctionType(
   llvm_unreachable("invalid representation");
 }
 
+const clang::Type *ClangTypeConverter::getFunctionType(
+    ArrayRef<SILParameterInfo> params, Optional<SILResultInfo> result,
+    SILFunctionType::Representation repr) {
+
+  // Using the interface type is sufficient as Swift does not allow abstracting
+  // over @convention(c) functions, hence we don't need any substitutions.
+  auto resultClangTy = result.hasValue()
+                     ? convert(result.getValue().getInterfaceType())
+                     : ClangASTContext.VoidTy;
+
+  if (resultClangTy.isNull())
+    return nullptr;
+
+  SmallVector<clang::FunctionProtoType::ExtParameterInfo, 4> extParamInfos;
+  SmallVector<clang::QualType, 4> paramsClangTy;
+  bool someParamIsConsumed = false;
+  for (auto &p : params) {
+    auto pc = convert(p.getInterfaceType());
```

Comment:
Does the same comment about generics apply here too? You might want to add an assert